core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/openapi.kt

```diff
@@ -0,0 +1,2173 @@
+             |}
+             |${generatedTypeAliases?.declarations?.replace("\n", "\n|") ?: ""}
+             |${generatedExtensionProperties?.declarations?.replace("\n", "\n|") ?: ""}
+         """.trimMargin()
+    }
+
+    return CodeWithConverter(
+        declarations = "", // Need to use the converter to get the right name of singleton object
+        converter = { getCode(ValidFieldName.of(it)) },
+    )
+}
+
+private interface IsObjectOrList {
+    val isObject: Boolean
+    val isList: Boolean
+}
+
+/**
+ * A [DataSchema] interface can implement this if it represents a map-like data schema (so key: value).
+ * Used in OpenAPI to represent objects with 'just' additionalProperties of a certain type.
+ */
+public interface AdditionalProperty<T> : KeyValueProperty<T> {
+
+    /** Key of the property. */
+    override val key: String
+
+    /** Value of the property. */
+    override val value: T
+
+    public companion object {
+        internal fun getMarker(typeArguments: List<String>) = Marker(
+            name = AdditionalProperty::class.qualifiedName!!,
+            isOpen = false,
+            fields = listOf(
+                generatedFieldOf(
+                    fieldName = ValidFieldName.of(AdditionalProperty<*>::key.name),
+                    columnName = AdditionalProperty<*>::key.name,
+                    overrides = false,
+                    fieldType = FieldType.ValueFieldType(String::class.qualifiedName!!),
+                ),
+                generatedFieldOf(
+                    fieldName = ValidFieldName.of(AdditionalProperty<*>::`value`.name),
+                    columnName = AdditionalProperty<*>::`value`.name,
+                    overrides = false,
+                    fieldType = FieldType.ValueFieldType(Any::class.qualifiedName!! + "?"),
+                ),
+            ),
+            superMarkers = emptyList(),
+            visibility = MarkerVisibility.EXPLICIT_PUBLIC,
+            typeParameters = emptyList(),
+            typeArguments = typeArguments,
+        )
+
+        /**
+         * Used to convert a [DataFrame] to an [AdditionalProperty] [DataFrame] by [DataFrame.gather]ing all column names
+         * into a column named `key` and the values into a column named `value` of type `Any?`.
+         *
+         * We first convert the columns (and recursively down the frame-tree) before gathering this
+         * DataFrame. This is to prevent `gather { all() }` from creating a lot of `null` values due to
+         * merging of ColumnGroups.
+         *
+         *
+         * @receiver the DataFrame to convert to an [AdditionalProperty] [DataFrame].
+         * @param convertTo optional [ConvertSchemaDsl] to specify extra conversions that need to occur.
+         * @return a [DataFrame] of [AdditionalProperty]s.
+         */
+        @Suppress("UNCHECKED_CAST")
+        public fun AnyFrame.convertToAdditionalProperties(
```

Comment:
maybe mention that this used in generated code?

Reply:
sure